### PR TITLE
Update room hud

### PR DESCRIPTION
**Pull request by hoalex21**

### CODE REVIEW

File: frontend/components/leaderboard.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        if (gameState.players) {
+            setPlayers(gameState.players)
+        }
+    }, [gameState])
+
+    return (
+        <>
+            <p className="h2">Leaderboard</p>
+            <ul>
```
Comment (devappslabs) [inline frontend/components/leaderboard.tsx:21]: let's display the leaderboard in order of who has the most points

File: frontend/components/round.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+"use client";
+
+import { useEffect, useState } from "react"
+
+export default function Round({ gameState }) {
```
Comment (ll886) [inline frontend/components/round.tsx:5]: it doesn't look like the round updates after each round.

I think maybe pass in the socket and call `setRound` everytime a `updateGameState` event happens

**Review by devappslabs — CHANGES_REQUESTED** [review]

File: frontend/app/rooms/[roomId]/page.tsx [hunk 3]
```diff
@@ -17,11 +20,13 @@ function Page() {
   const [isLinkCopied, setIsLinkCopied] = useState(false);
   const [playerId, setPlayerId] = useState(null);
   const [gameState, setGameState] = useState<Game | null>(null);
+  const [drawWordInfo, setDrawWordInfo] = useState<string>("");
 
   useEffect(() => {
     generateGuestIdIfNull();
     socket.connect();
     console.log("connecting socket...");
+    generateGuestIdIfNull();
```
Comment (devappslabs) [inline frontend/app/rooms/[roomId]/page.tsx]: remove this duplicate line

**Review by devappslabs — CHANGES_REQUESTED** [review]

File: frontend/components/leaderboard.tsx (hunk 1 above)
Comment (hoalex21) [reply frontend/components/leaderboard.tsx:21]: done

File: frontend/app/rooms/[roomId]/page.tsx (hunk 3 above)
Comment (hoalex21) [reply frontend/app/rooms/[roomId]/page.tsx]: done

**Review by devappslabs — APPROVED** [review]

**Pull request merged**